### PR TITLE
Add providers list to the leftnav when on registries file route

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: []
-   Feature flag: n/a

## Purpose
- Show file-providers for registries

## Summary of Changes
- Add element to show a registration's files provider on the registration overview page when you are on the registration's files route
- Change `inverse` for the `target` relationship of the file/file-provider models as this was incorrectly adding files to the `registration.files` relationship

## Screenshot(s)
image

### CODE REVIEW

File: lib/registries/addon/overview/template.hbs [hunk 1]
```diff
@@ -62,8 +62,33 @@
...
-                    <leftNav.link data-analytics-name='Files' @route='registries.overview.files' @icon='file-alt'
-                        @label={{t 'registries.overview.external_links.files'}} />
+                    <leftNav.link data-analytics-name='Files' @route='registries.overview.files'
+                        @models={{array this.registration.id}} @icon='file-alt'
+                        @label={{t 'registries.overview.files.title'}} />
+                    {{#if this.onFilesRoute}}
+                        <div local-class='FileProvidersList'>
+                            {{#each this.registration.files as |provider|}}
+                                <div local-class='FileProvider'>
+                                    {{provider.name}}
```
Comment (brianjgeiger) [inline lib/registries/addon/overview/template.hbs:72]: I think we're going to have to make some way of converting the id to a display name. It should show as "OSF storage" not "osfstorage". Maybe we could use our translation service, and just pass that in as a key?

File: lib/registries/addon/overview/template.hbs [hunk 2]
```diff
@@ -62,8 +62,33 @@
...
+                                            <FaIcon
+                                                @icon='globe'
+                                            >
+                                            </FaIcon>
+                                            <BsTooltip
+                                                @placement='right'
+                                                @triggerEvents='hover'
+                                            >
+                                                {{t 'registries.overview.files.storage_location'}}
+                                                {{!-- region here --}}
```
Comment (brianjgeiger) [inline lib/registries/addon/overview/template.hbs:84]: You should be able to get this from the registration's region relationship, but you'll have to add it to the model, it looks like. There's no separate story ticket for this, but we could add one if you don't think it's appropriate for this PR.

File: lib/registries/addon/overview/template.hbs [hunk 3]
```diff
@@ -62,8 +62,33 @@
...
+                            {{#each this.registration.files as |provider|}}
+                                <div local-class='FileProvider'>
+                                    {{provider.name}}
+                                    {{#if (eq provider.name 'osfstorage')}}
+                                        <span>
+                                            <FaIcon
+                                                @icon='globe'
+                                            >
+                                            </FaIcon>
+                                            <BsTooltip
```
Comment (brianjgeiger) [inline lib/registries/addon/overview/template.hbs:79]: Will we be able to style this to look like the mockups, or should we verify that everyone's cool with a standard tooltip?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Overall, looks good. Just a few comments.

File: app/models/file.ts [hunk 4]
```diff
@@ -50,9 +50,8 @@ export default class FileModel extends BaseFileItem {
     @hasMany('comment', { inverse: null })
     comments!: AsyncHasMany<CommentModel>;
 
-    // TODO: In the future apiv2 may also need to support this pointing at nodes OR registrations
```
Comment (brianjgeiger) [inline app/models/file.ts:53]: Look at you, fulfilling this TODO.

**Review by brianjgeiger — APPROVED** [review]
Lookin' good.

File: app/models/file.ts (hunk 4 above)
Comment (futa-ikeda) [reply app/models/file.ts:53]: Consider it TO-DONE 😄

File: lib/registries/addon/overview/controller.ts [hunk 5]
```diff
@@ -43,6 +43,11 @@ export default class Overview extends Controller {
         return true;
     }
 
+    @computed('router.currentRouteName')
```
Comment (adlius) [inline lib/registries/addon/overview/controller.ts:46]: Do we really need the `@computed` decorator here?

**Pull request merged**